src/lib.rs

```diff
@@ -260,26 +273,54 @@ impl Watch {
     }
 }
 
+#[non_exhaustive]
 #[derive(Debug, StructOpt)]
 pub struct DevServer {
     #[structopt(long, default_value = "127.0.0.1")]
-    ip: IpAddr,
+    pub ip: IpAddr,
     #[structopt(long, default_value = "8000")]
-    port: u16,
+    pub port: u16,
 
     #[structopt(flatten)]
-    watch: Watch,
+    pub watch: Watch,
+    #[structopt(skip)]
+    pub served_path: Option<PathBuf>,
 }
 
 impl DevServer {
-    pub fn serve(&self, build_dir_path: impl AsRef<Path>) -> Result<()> {
+    pub fn serve(&self) -> Result<()> {
         let address = SocketAddr::new(self.ip, self.port);
         let listener = TcpListener::bind(&address).context("cannot bind to the given address")?;
 
-        log::info!("Development server at: http://{}", &address);
+        log::info!("DevServer: Development server at: http://{}", &address);
+
+        let served_path = if let Some(path) = &self.served_path {
+            path.to_owned()
+        } else {
+            let metadata = metadata();
+
+            if metadata
+                .target_directory
+                .join("release")
+                .join("dist")
+                .exists()
```

Comment:
You can't rely on the existence of the path to know if it needs to be served or not. We already had tons of issues because of that.